services/system/AuthSig/plugin/src/lib.rs

```diff
@@ -1,28 +1,46 @@
 #[allow(warnings)]
 mod bindings;
-
-use bindings::auth_sig::plugin::types::Keypair;
-use bindings::host::common::server as Server;
-use bindings::host::common::types as CommonTypes;
+use bindings::auth_sig::plugin::types::{Keypair, Pem};
+use bindings::clientdata::plugin::keyvalue as Keyvalue;
 use bindings::exports::auth_sig::plugin::smart_auth;
 use bindings::exports::auth_sig::plugin::{
     actions::Guest as Actions, keyvault::Guest as KeyVault, smart_auth::Guest as SmartAuth,
 };
+use bindings::host::common::types as CommonTypes;
+use bindings::transact::plugin::intf as Transact;
 
 mod errors;
 use errors::ErrorType::*;
 
 use p256::ecdsa::{SigningKey, VerifyingKey};
-use p256::pkcs8::Document;
-use p256::pkcs8::{EncodePrivateKey, EncodePublicKey, LineEnding};
+use p256::pkcs8::{DecodePrivateKey, EncodePrivateKey, EncodePublicKey, LineEnding};
 use rand_core::OsRng;
 
 use psibase::fracpack::Pack;
 use psibase::services::auth_sig::action_structs as MyService;
 
-struct Component;
+use md5;
+
+trait TryFromPemStr: Sized {
+    fn try_from_pem_str(p: &Pem) -> Result<Self, CommonTypes::Error>;
+}
+
+impl TryFromPemStr for pem::Pem {
+    fn try_from_pem_str(key_string: &Pem) -> Result<Self, CommonTypes::Error> {
+        Ok(pem::parse(key_string.trim()).map_err(|e| CryptoError.err(&e.to_string()))?)
+    }
+}
+
+// Uses md5 because it's short, this doesn't need to be cryptographically secure.
```

Comment:
In that case it would be better to use a non-cryptographic hash function rather than a known-insecure cryptographic one.

Reply:
Changed to seahash